File: simplified-books-core-ng/src/main/java/org/nypl/simplified/books/accounts/AccountsDatabases.java
package org.nypl.simplified.books.accounts;

import org.nypl.simplified.books.book_database.BookDatabaseFactoryType;
import org.nypl.simplified.books.book_database.BookDatabases;

import java.io.File;

public final class AccountsDatabases implements AccountsDatabaseFactoryType {

  private static final AccountsDatabases INSTANCE = new AccountsDatabases();

  public static AccountsDatabases get() {
    return INSTANCE;
  }

  private AccountsDatabases() {

  }

  @Override
  public AccountsDatabaseType openDatabase(
      final BookDatabaseFactoryType book_databases,
      final AccountProviderCollectionType account_providers,
      final File directory)
      throws AccountsDatabaseException {
    return AccountsDatabase.open(book_databases, account_providers, directory);
  }

  @Override
  public AccountsDatabaseType openDatabase(
      final AccountProviderCollectionType account_providers,
      final File directory)
      throws AccountsDatabaseException {
    return AccountsDatabase.open(BookDatabases.get(), account_providers, directory);
  }
}

File: simplified-books-core-ng/src/main/java/org/nypl/simplified/books/accounts/AccountsDatabases.kt
package org.nypl.simplified.books.accounts

import android.content.Context
import org.nypl.simplified.books.book_database.BookDatabaseFactoryType
import org.nypl.simplified.books.book_database.BookDatabases
import org.nypl.simplified.observable.ObservableType
import java.io.File

/**
 * The default implementation of the {@link AccountsDatabaseFactoryType} interface.
 */

object AccountsDatabases : AccountsDatabaseFactoryType {

  @Throws(AccountsDatabaseException::class)
  override fun openDatabase(
    context: Context,
    accountEvents: ObservableType<AccountEvent>,
    bookDatabases: BookDatabaseFactoryType,
    accountProviders: AccountProviderCollectionType,
    directory: File): AccountsDatabaseType {
    return AccountsDatabase.open(context, accountEvents, bookDatabases, accountProviders, directory)
  }

  @Throws(AccountsDatabaseException::class)
  override fun openDatabase(
    context: Context,
    accountEvents: ObservableType<AccountEvent>,
    accountProviders: AccountProviderCollectionType,
    directory: File): AccountsDatabaseType {
    return AccountsDatabase.open(context, accountEvents, BookDatabases, accountProviders, directory)
  }
}
